src/coreComponents/mesh/mpiCommunications/mpiCommunicationUtils.hpp

```diff
@@ -0,0 +1,133 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file CommID.hpp
+ */
+
+#ifndef GEOSX_MESH_MPICOMMUNICATIONS_MPICOMMUNICATIONUTILS_HPP_
+#define GEOSX_MESH_MPICOMMUNICATIONS_MPICOMMUNICATIONUTILS_HPP_
+
+#include "common/DataTypes.hpp"
+
+namespace geosx
+{
+
+struct SyncFieldsID
+{
+  void addFields(FieldLocation const location, std::vector<string> const fieldNames )
+  {
+    string const key = generateKey(location);
+    addFields(fieldNames, key);
+  }
+
+  void addFields(FieldLocation const location, std::vector<string> const fieldNames, std::vector<string> const regionNames )
+  {
+    for (auto const & regionName : regionNames)
+    {
+      string const key = generateKey(location, regionName);
+      addFields(fieldNames, key);
+    }
+  }
+
+  std::map< string, array1d<string> > const & getFields() const
+  {
+    return m_fields;
+  }
+  
+  static string const getRegionName( string const & key )
+  {
+    string regionName(key);
+    regionName.erase(0, 5);
+    return regionName;
+  }
+
+  static FieldLocation const getLocation( string const & key )
+  {
+    FiedlLocation location;
+    if (key.contains("nodes"))
```

Comment:
the strings should be stored specified in a `static constexpr char const * ` like all the viewkeystructs...which coincidently is now a terrible name